src/envs/__init__.py

```diff
@@ -27,27 +31,39 @@
 ]
 
 
+_MOST_RECENT_ENV_INSTANCE = {}
+
 def create_env(name: str) -> BaseEnv:
-    """Create an environment given its name.
-    """
+    """Create an environment given its name."""
     if name == "cover":
-        return CoverEnv()
-    if name == "cover_typed_options":
-        return CoverEnvTypedOptions()
-    if name == "cover_hierarchical_types":
-        return CoverEnvHierarchicalTypes()
-    if name == "cover_multistep_options":
-        return CoverMultistepOptions()
-    if name == "cluttered_table":
-        return ClutteredTableEnv()
-    if name == "blocks":
-        return BlocksEnv()
-    if name == "painting":
-        return PaintingEnv()
-    if name == "playroom":
-        return PlayroomEnv()
-    if name == "behavior":
-        return BehaviorEnv()  # pragma: no cover
-    if name == "repeated_nextto":
-        return RepeatedNextToEnv()
-    raise NotImplementedError(f"Unknown env: {name}")
+        env = CoverEnv() # type: ignore
+    elif name == "cover_typed_options":
+        env = CoverEnvTypedOptions() # type: ignore
+    elif name == "cover_multistep_options":
+        env = CoverMultistepOptions() # type: ignore
+    elif name == "cover_hierarchical_types":
+        env = CoverEnvHierarchicalTypes() # type: ignore
+    elif name == "cluttered_table":
+        env = ClutteredTableEnv() # type: ignore
+    elif name == "blocks":
+        env = BlocksEnv() # type: ignore
+    elif name == "painting":
+        env = PaintingEnv() # type: ignore
+    elif name == "behavior":
+        env = BehaviorEnv() # type: ignore # pragma: no cover
+    elif name == "playroom":
+        env = PlayroomEnv() # type: ignore
+    elif name == "repeated_nextto":
+        return RepeatedNextToEnv() # type: ignore
+    else:
+        raise NotImplementedError(f"Unknown env: {name}")
+
+    _MOST_RECENT_ENV_INSTANCE[name] = env
+    return env
+
+
+def get_env_instance(name: str) -> BaseEnv: # pragma: no cover
```

Comment:
we should cover this in a unit test. in fact, would you mind splitting off the changes to this file (except for the behavior env lines) and creating a small PR with just that?

Reply:
yup, will open one!

Reply:
addressed in #248